core/tile-entity/init.lua

```diff
@@ -0,0 +1,63 @@
+local component = require("component")
+local event = require("event")
+
+---@class tileEntity
+---@field address string
+---@field position coordinates
+local tileEntity = {}
+
+local knownEntities = {}
+local knownEntityTypes = {}
+
+function tileEntity.addType(componentName, description)
+    knownEntityTypes[componentName] = description or componentName
+end
+
+function tileEntity.listKnownEntities() return pairs(knownEntities) end
+
+local function isKnownType(componentType)
+    for type in pairs(knownEntityTypes) do
+        if componentType == type then return true end
+    end
+end
+
+function tileEntity.refreshKnownEntitiesList()
+    for address, componentType in component.list() do
+        if isKnownType(componentType) then
+            knownEntities[address] = component.proxy(address)
+        end
+    end
+end
+
+---Creates a new tileEntity object
+---@param address string
+---@param position coordinates
```

Comment:
What is the format for `coordinates`? {x, y, z}, {x=number, y=number, z=number}? Something else?

Reply:
The type is defined in the [types.lua](https://github.com/S4mpsa/NIDAS/blob/254e4bc796b5039318d5c09cf350cf24dbef32c7/core/tile-entity/types.lua) file

Reply:
I see this is in `types.lua`